docs/reference/analysis/tokenfilters/keyword-marker-tokenfilter.asciidoc

```diff
@@ -1,137 +1,189 @@
 [[analysis-keyword-marker-tokenfilter]]
-=== Keyword Marker Token Filter
+=== Keyword marker token filter
+++++
+<titleabbrev>Keyword marker</titleabbrev>
+++++
 
-Protects words from being modified by stemmers. Must be placed before
-any stemming filters.
+Excludes specified tokens, called <<stemmer-keyword,keywords>>, from
+<<stemming,stemming>>.
 
-[cols="<,<",options="header",]
-|=======================================================================
-|Setting |Description
-|`keywords` |A list of words to use.
+The `keyword_marker` filter is often used with custom
+<<analysis-lang-analyzer,language analyzers>> to protect specific words from
+stemming. This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/miscellaneous/KeywordMarkerFilter.html[KeywordMarkerFilter].
 
-|`keywords_path` |A path (either relative to `config` location, or
-absolute) to a list of words.
+[TIP]
+====
+If you want to preserve both a stemmed and unstemmed version of each token,
+we recommend using the <<analysis-keyword-repeat-tokenfilter,`keyword_repeat`>>
+token filter.
+====
 
-|`keywords_pattern` |A regular expression pattern to match against words
-in the text.
+[[analysis-keyword-marker-tokenfilter-analyze-ex]]
+==== Example
 
-|`ignore_case` |Set to `true` to lower case all words first. Defaults to
-`false`.
-|=======================================================================
-
-You can configure it like:
+The following <<indices-analyze,analyze API>> request uses the `keyword_marker`
+filter to protect the `jumping` keyword in `quick foxes running and jumping`
+from stemming:
```

Comment:
I think this example needs a bit more explanation--something about the request analyzing the specified text using a custom analyzer that designates _jumping_ as a keyword.

Reply:
The point here is to emphasize the input and output of the `keyword_marker` filter. The description above is _telling_ you how it works, this is _showin_ how it works.

I think labelling this as a transient custom analyzer, while accurate, could confuse the point a bit.

Could you elaborate a bit more on why the current sentence is inadequate?

Reply:
I agree it makes sense to put the emphasis on the filter input/output--maybe it's just that I had the context of the old and new example. I think I also tripped over the "protect from stemming" phrasing. To me, you designate a term as a keyword to prevent it from being stemmed, or maybe to skip stemming? Feel free to stet if you prefer it as-is.

Reply:
Thanks! I've updated this sentence with 9a766c9be8d.